core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -318,28 +320,47 @@ private void failAndRespond(Throwable cause, AggregatedHttpMessage message, Http
         final HttpData content = message.content();
 
         logBuilder().responseHeaders(headers);
+        logBuilder().increaseResponseLength(content.length());
 
         final State state = this.state; // Keep the state before calling fail() because it updates state.
-        fail(cause);
+        setDone();
```

Comment:
Could you explain why we need to change this?

Reply:
The previous `fail()` method is as follows:
```
    private void fail(Throwable cause) {
        setDone();
        logBuilder().endResponse(cause);
        subscription.cancel();  
    }
```
As you know, the last line was added yesterday by rag-san. 
I wanted to postpone calling `logBuilder().endResponse(cause)` after finishing write operation so that I can set the root cause to the log builder. So I removed `fail()` and called  `setDone()` and `logBuilder().endResponse(cause)` separately.

Reply:
OK! Thanks for explanation :-)